src/measure_theory/decomposition/signed_hahn.lean

```diff
@@ -0,0 +1,682 @@
+/-
+Copyright (c) 2021 Kexing Ying. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kexing Ying
+-/
+import measure_theory.vector_measure
+import order.symm_diff
+
+/-!
+# Hahn decomposition
+
+This file defines positive and negative sets with respect to a vector measure and
+prove the Hahn decomposition theorem (signed version).
+
+## Main definitions
+
+* `vector_measure.positive` : a set `i` is positive with respect to a vector measure `v` if `v`
+  restricted on `i` is non-negative, i.e. `0 ≤ v.restrict i`.
+* `vector_measure.negative` : a set `i` is negative with respect to a vector measure `v` if `v`
+  restricted on `i` is non-positive, i.e. `v.restrict i ≤ 0`.
+
+## Main results
+
+* `signed_measure.exists_disjoint_positive_negative_union_eq` : the Hahn decomposition theorem.
+* `signed_measure.exists_negative_set` : A measurable set of negative measure contains at least
+  one negative subset.
+
+## Tags
+
+positive, negative, Hahn decomposition theorem
+-/
+
+noncomputable theory
+open_locale classical big_operators nnreal ennreal
+
+section lemmas
+
+variables {α : Type*}
+
+open set filter
+
+lemma exists_tendsto_Inf {S : set ℝ} (hS : ∃ x, x ∈ S) (hS' : ∃ x, ∀ y ∈ S, x ≤ y) :
```

Comment:
Can you state this for a general `conditionally_complete_linear_order`?

Reply:
Is this true for  `conditionally_complete_linear_order`? I'm thinking that if $[0, 1) \cup 2$ equipped with the real ordering form a `conditionally_complete_linear_order` and the set $[0, 1)$ is a nonempty bounded set with supremum = 2 but it doesn't have a sequence converging to it.

Reply:
You need the topology to be the `order_topology` and to also be `first_countable`, and then everything is fine.